imgpkg/002-recursive-bundles/README.md

```diff
@@ -0,0 +1,280 @@
+    6. The Consumer retrieves the Bundle generated by the Packager
+    7. The Consumer retrieves every other Bundle that is required
+
+    This approach has challenges for both Packager and Consumer
+    The Consumer would have to download multiple separate bundles and follow the instructions from the Packager to install all the applications
+    The Packager would have to collect the information about the applications and generate an installation guide for the Consumer.
+
+### Mitigate Pain Points
+
+In this proposal, we adapt `imgpkg` to try to mitigate some of the current pains.
+Given that `imgpkg` already supports [bundles](https://carvel.dev/imgpkg/docs/latest/resources/#bundle) with a simple extension that allows bundles to contain other bundles can be created to help with the problem.
+
+The workflow that this change enables is the following:
+1. Each Application Team generates a bundle with all the needed images and the configuration
+2. The Packager would collect the applications Image Bundle SHA's for all applications
+3. The Packager can provide some overlay's for the configuration and add some utilities that are used during installation
+4. Create a Bundle that contains all application Image Bundles artifacts generated in the previous point
+5. The Application Consumer would be given a single Bundle Image that can be downloaded and consumed
+
+The benefits of this approach versus the current state are
+- Each Application Team manages their application's Bundle. The team that builds the application will manage all the requirements
+- The Packager just needs to collect the Bundle SHA for each application
+- The Consumer can get all the needed resources from a single Bundle to install all the applications needed
+
+# Proposed changes
+
+In this section, the proposal will only talk about parts that will be changed based on the Recursive Bundle solution
+
+## Pushing a bundle
+
+The major proposed change in this section is to remove the validation done today that checks if the bundle being pushed contains bundles in the `.imgpkg/images.yml` Lock file.
+
+### Required changes
+
+#### Validation removal
+
+Today we validate to ensure that when we are pushing a bundle, the images associated with it are not bundles.
+The error message that is shown read `Error: Expected image lock to not contain bundle reference:`
+
+As part of this proposal, this check no longer will be done.
+
+#### Ignore bundle specific folder
+
+When a bundle is pulled, and contain recursive bundles, the folder `.bundles` is created in the root of the bundle.
+As part of this proposal, we should ensure that if a user tries to create a bundle and it contains a folder called `.bundles`
+`imgpkg` should ignore the folder and provide the following message
+
+```
+Warning: .bundles folder could not be added to the bundle.
+         This folder is used to store configurations of recursive bundles and cannot be checked into the bundle
```

Comment:
What do you think of shorter warning like `Warning: Ignoring ".imgpkg/bundles" directory (directory is only used for keeping pulled dependent bundles)`?